subprojects/docs/src/docs/userguide/lazyConfiguration.adoc

```diff
@@ -15,40 +15,226 @@
 
 [NOTE]
 ====
 
-The provider API is currently <<feature_lifecycle,incubating>>. Please be aware that the DSL and other configuration may change in later Gradle versions.
+The Provider API is currently <<feature_lifecycle,incubating>>. Please be aware that the DSL and other configuration may change in later Gradle versions.
 
 ====
 
-[[sec:creating_new_property_instances]]
-=== Creating new instances
+First, Gradle provides lazy properties, which delay the calculation of a property’s value until it’s absolutely required. This provides two main benefits to build script and plugin authors:
+
+1. Build authors can wire together Gradle models without worrying when a particular property’s value will be known.
+2. Build authors can avoid resource intensive work during the configuration phase, which can have a direct impact on maximum build performance.
+
+Gradle represents lazy properties with two interfaces:
+
+* api:org.gradle.api.provider.Provider[] are properties that can only be queried and cannot be changed.
+** Properties with these types are read-only.
+** The method api:org.gradle.api.provider.Provider#get()[] returns the current value of the property.
+** A `Provider` can be created by the factory method api:org.gradle.api.Project#provider(java.util.concurrent.Callable)[].
+* api:org.gradle.api.provider.Property[] are properties that can be queried and overwritten.
+** Properties with these types are configurable.
+** `Property` implements the `Provider` interface.
+** The method api:org.gradle.api.provider.Property#set(T)[] provides a value for the property, overwriting whatever value may have been present.
+** A `Property` can be created by the factory method api:org.gradle.api.model.ObjectFactory#property(java.lang.Class)[].
+
+Neither of these types nor their subtypes are intended to be implemented by a build script or plugin author.  Gradle provides several factory methods to create instances of these types. See the <<lazy_configuration_reference,Quick Reference>> for all of the types and factories available.
+
+Lazy properties are intended to be passed around and only evaluated when required (usually, during the execution phase). For more information about the Gradle build phases, please see <<sec:build_phases>>.
+
+The following demonstrates a task with a read-only property and a configurable property:
+
+++++
+<sample id="usePropertyAndProvider" dir="providers/propertyAndProvider" title="Using a read-only and configurable property">
+    <sourcefile file="build.gradle"/>
+    <output args="greeting"/>
+</sample>
+++++
+
+The `Greeting` task has a `Property<String>` for the mutable part of the message and a `Provider` for the modified, read-only, message.
+
+[NOTE]
+====
+
+Note that Groovy Gradle DSL will generate setter methods for each `Property`-typed property in a task implementation. These setter methods allow you to configure the property using the assignment (`=`) operator as a convenience.
+
+====
+
+[[sec:why_new_types]]
```

Comment:
I think a section to explain this is too much. Perhaps 1 sentence in the introduction to relate this concept to Convention Mapping (with a link, if possible)

Reply:
Maybe just move this into FAQs?